src/js/sequences.js

```diff
@@ -0,0 +1,306 @@
+/**
+ * SEQUENCING GENERATING FUNCTIONS
+ */
+
+/* global alert, location, jQuery, localStorage, navigator */
+import { PRIMES } from './constants.js'
+import { line_to_decimal, decimal_to_cents } from './helpers.js'
+import { sum_array, get_prime_limit } from './numbers.js'
+
+// calculate a continued fraction for the given number
+function get_cf(num, maxiterations, roundf) {
+  var cf = [] // the continued fraction
+  var digit;
+
+  var roundinv = 1.0 / roundf;
+
+  var iterations = 0;
+  while (iterations < maxiterations) {
+    digit = Math.floor(num);
+    cf.push(digit);
+
+    num -= digit;
+
+    if (num === 0 || num <= roundinv) {
+      break;
+    }
+
+    num = 1.0 / num;
+    iterations++;
+  }
+
+  return cf;
+}
+
+// calculate a single convergent for a given continued fraction
+function get_convergent(cf, depth=0) {
+  var cfdigit; // the continued fraction digit
+  var num; // the convergent numerator
+  var den; // the convergent denominator
+  var tmp; // for easy reciprocation
+
+  if (depth >= cf.length || depth === 0)
+    depth = cf.length;
+
+  for (let d = 0; d < depth; d++) {
+    cfdigit = cf[d];
+    num = cfdigit;
+    den = 1;
+
+    // calculate the convergent
+    for (let i = d; i > 0; i--) {
+    tmp = den;
+    den = num;
+    num = tmp;
```

Comment:
This can also be simplified with the destructuring operator to `[den, num] = [num, den]`

Reply:
Will be rewriting this algorithm soon